apps/render_cf.py

```diff
@@ -192,45 +192,63 @@ def render_templates(job_types, compute_envs, security_environment, api_name):
         template_file.with_suffix('').write_text(output)
 
 
-def get_compute_environments(job_types: dict, compute_env_file: Optional[Path]) -> list[dict]:
+def parse_compute_environments_file(
+    compute_env_names: set,
+    compute_env_imports: set,
+    compute_env_file: Path
+) -> list[dict]:
     compute_envs = []
-    compute_env_names = set()
+    compute_envs_from_file = yaml.safe_load(compute_env_file.read_text())['compute_environments']
+
+    for name in compute_envs_from_file:
+        if name in compute_env_imports:
+            if name in compute_env_names:
+                raise ValueError(
+                    f'Compute envs must have unique names but the following is defined more than once: {name}.'
+                )
+            compute_envs_from_file[name].update({'name': name})
+            compute_envs.append(compute_envs_from_file[name])
+            compute_env_names.add(name)
```

Comment:
Should it be more like this? So that the error gets raised for duplicate names even if they're not imported?
```suggestion
    for name in compute_envs_from_file:
        if name in compute_env_names:
            raise ValueError(
                f'Compute envs must have unique names but the following is defined more than once: {name}.'
            )
        if name in compute_env_imports:
            compute_envs_from_file[name].update({'name': name})
            compute_envs.append(compute_envs_from_file[name])
        compute_env_names.add(name)
```

Reply:
I did it like that intentionally so that they could be overridden by different deployments, which seems in line with how a lot of other things work. I feel like there was another reason I did that too but I can't remember.

Reply:
Hm OK, let me read it more carefully

Reply:
Yep, I think that makes sense.

Reply:
Honestly, I'll probably change it to your suggestion. I don't really love that behavior the more I think about it. I'm just playing around with it to see if I remember what the other issue was.